tests/e2e/utils/KubernetesCommandLineToolsExecutor.ts

```diff
@@ -0,0 +1,131 @@
+        return this.namespace;
+    }
+
+    loginToOcp(): void {
+        if (this.KUBERNETES_COMMAND_LINE_TOOL === KubernetesCommandLineTool.OC) {
+            Logger.debug(`${this.getLoggingName(this.loginToOcp.name)}: Login to the \'OC\' client`);
+            const url: string = this.getServerUrl();
+            Logger.debug(url, TestConstants.TS_SELENIUM_OCP_USERNAME);
+            exec(`sleep 5
+            oc login --server=${url} -u=${TestConstants.TS_SELENIUM_OCP_USERNAME} -p=${TestConstants.TS_SELENIUM_OCP_PASSWORD} --insecure-skip-tls-verify`);
+        } else {
+            Logger.debug(`${this.getLoggingName(this.loginToOcp.name)}: Kubernetes command line tool not "openshift client"`);
+        }
+    }
+
+    getContainerName(): string {
+        Logger.debug(`${this.getLoggingName(this.getContainerName.name)}: Get container name.`);
+        const output: ShellString = this.execWithLog(`${(this.KUBERNETES_COMMAND_LINE_TOOL)} get ${(KubernetesCommandLineToolsExecutor.pod)} -o jsonpath='{.spec.containers[*].name}' -n ${this.namespace}`);
+        echo('\n');
+        return output.stderr ? output.stderr : output.stdout;
+    }
+
+    getWorkspacePodName(): string {
+        Logger.debug(`${this.getLoggingName(this.getWorkspacePodName.name)}: Get workspace pod name.`);
+        const output: ShellString = this.execWithLog(`${(this.KUBERNETES_COMMAND_LINE_TOOL)} get pod -l controller.devfile.io/devworkspace_name=${this.workspaceName} -n ${this.namespace} -o name`);
+        return output.stderr ? output.stderr : output.stdout.replace('\n', '');
+    }
+
+    deleteDevWorkspace(): void {
+        Logger.debug(`${this.getLoggingName(this.deleteDevWorkspace.name)}: Delete '${this.workspaceName}' workspace`);
+        this.execWithLog(`${(this.KUBERNETES_COMMAND_LINE_TOOL)} patch dw ${this.workspaceName} -n ${this.namespace} -p '{ "metadata": { "finalizers": null }}' --type merge || true`);
+        this.execWithLog(`${(this.KUBERNETES_COMMAND_LINE_TOOL)} delete dw ${this.workspaceName} -n ${this.namespace} || true`);
+    }
+
+    applyAndWaitDevWorkspace(yamlConfiguration: string): ShellString {
+        if (this.KUBERNETES_COMMAND_LINE_TOOL === KubernetesCommandLineTool.KUBECTL) {
+            this.createNamespace();
+        }
+        this.apply(yamlConfiguration);
+        const output: ShellString = this.waitDevWorkspace();
+        KubernetesCommandLineToolsExecutor.pod = this.getWorkspacePodName();
+        KubernetesCommandLineToolsExecutor.container = this.getContainerName();
+        return output;
+    }
+
+    executeCommand(commandToExecute: string): ShellString {
+        Logger.debug(`${this.getLoggingName(this.executeCommand.name)}:`);
+        return this.execWithLog(`${(this.KUBERNETES_COMMAND_LINE_TOOL)} exec -i ${KubernetesCommandLineToolsExecutor.pod} -n ${this.namespace} -c ${KubernetesCommandLineToolsExecutor.container} -- sh -c "${commandToExecute}"`);
+    }
+
+    apply(yamlConfiguration: string): void {
```

Comment:
We can apply the kube objects in different ways:
* from output (`cat <<EOF | oc apply -n ${this.namespace} -f - ....`)
* from external URL (`oc apply -f https://gist.githubusercontent.com`)
* from a file system using path (`oc apply -f <path to yaml file>`)
Is it make sense to rename applyUsingOutput() or something like this?

Reply:
okay